libr/asm/arch/8051/8051_ass.c

```diff
@@ -1040,6 +1073,31 @@ static bool mnem_mov(char const*const*arg, ut16 pc, ut8**out) {
 		*out += 3;
 		return true;
 	}
+
+	if (!r_str_casecmp (arg[0], "@r0")) {
```

Comment:
the disasm strnig should be always in lowercase, you can lower it with `r_str_ucase` before entering here. 

Another option to handle more regs than just @r0 is to check for @r && isdigit(arg[2])